articles/hibernate-one-to-one-mapping/index.md

```diff
@@ -0,0 +1,535 @@
+- @JoinColumn(name = "sub_id") - Tells hibernate to look for column sub_id in the customer table and use the information to find the appropriate subscription for the customer.
+
+ After creating the customer class create a subscription class with the following details.
+ - id
+ - name
+ - price
+ - description
+ - constructor without the id
+ - toString method
+
+ ```java
+@Entity
+@Table(name = "subscription")
+public class Subscription {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private int id;
+
+    @Column(name = "name")
+    private String name;
+
+    @Column(name = "price")
+    private Double price;
+
+    @Column(name = "description")
+    private String descrition;
+
+
+    @OneToOne(cascade = {
+            CascadeType.MERGE,
+            CascadeType.PERSIST,
+            CascadeType.REFRESH,
+            CascadeType.DETACH
+    },mappedBy = "subscription")
+    private Customer customer;
+
+    public Subscription(String name, Double price, String descrition) {
+        this.name = name;
+        this.price = price;
+        this.descrition = descrition;
+    }
+
+    public Subscription() {
+
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public Double getPrice() {
+        return price;
+    }
+
+    public void setPrice(Double price) {
+        this.price = price;
+    }
+
+    public String getDescrition() {
+        return descrition;
+    }
+
+    public void setDescrition(String descrition) {
+        this.descrition = descrition;
+    }
+
+    public Customer getCustomer() {
+        return customer;
+    }
+
+    public void setCustomer(Customer customer) {
+        this.customer = customer;
+    }
+
+    @Override
+    public String toString() {
+        return "Package{" +
+                "id=" + id +
+                ", name='" + name + '\'' +
+                ", price=" + price +
+                ", descrition='" + descrition + '\'' +
+                '}';
+    }
+}
+ ```
+ The CascadeType.REMOVE has been left out to avoid deleting the customer once a subscription is deleted to demonstrate cascading in action.
+
+mappedBy = "subscription" - added to mark the relationship as bi-directional meaning when a subscription is loaded the customer can be identified and vice versa.
+
+## Step 4: Creating and executing the main app
+---
+In this stage, the following will be covered.
+- Creating a new customer.
+- Retrieving customer by id and assigning a new subscription to the customer using a setter method.
+- Retrieving customer by id and deleting to see cascade delete applied to the subscription.
+- Retrieve and update a customer using a setter method.
+
+##### create a new customer
+---
+```java
+import com.javadev.hibernate.demo.entity.Customer;
```

Comment:
tech review stopped here. Please show how to setup project (in intellij) so that the reader does not have to scour around the internet to figure out how to import hibernate dependencies into their development environment. While some readers may already know this, your article is aimed at a general audience that very likely would not.

This should be included in your "Setting up the development environment" section.